dao/src/test/java/org/thingsboard/server/dao/service/entityprofile/EntityProfileServiceSqlTest.java

```diff
@@ -0,0 +1,126 @@
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.page.PageData;
+import org.thingsboard.server.common.data.page.PageLink;
+import org.thingsboard.server.dao.entityprofile.EntityProfileService;
+import org.thingsboard.server.dao.service.AbstractServiceTest;
+import org.thingsboard.server.dao.service.DaoSqlTest;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.junit.Assert.*;
+
+@DaoSqlTest
+public class EntityProfileServiceSqlTest extends AbstractServiceTest {
+    @Autowired
+    private EntityProfileService profileService;
+
+    @Test
+    public void testFindEntityProfilesByTenantId() {
+        EntityProfile saved = saveRandomEntityProfile();
+        profileService.findEntityProfilesByTenantId(saved.getTenantId()).stream()
+                .filter(saved::equals)
+                .findAny().orElseThrow(AssertionError::new);
+    }
+
+    @Test
+    public void testFindEntityProfilesByTenantIdWithPageLink() {
+        List<EntityProfile> all = profileService.findEntityProfilesByTenantId(TenantId.SYS_TENANT_ID);
+        PageLink pageLink = new PageLink(all.size());
+        PageData<EntityProfile> page = profileService.findEntityProfilesByTenantId(TenantId.SYS_TENANT_ID, pageLink);
+        assertEquals(new HashSet<>(all), new HashSet<>(page.getData()));
+    }
+
+    @Test
+    public void testFindEntityProfilesByTenantIdAndType() {
+        EntityProfile saved = saveRandomEntityProfile(randomAlphanumeric(10), EntityType.ASSET);
+        PageLink pageLink = new PageLink(1);
+        List<EntityProfile> data = profileService.findEntityProfilesByTenantIdAndType(saved.getTenantId(),
+                saved.getEntityType(), pageLink).getData();
+        assertEquals(pageLink.getPageSize(), data.size());
+        assertEquals(saved, data.get(0));
+    }
+
+    @Test
+    public void testFindTenantProfiles() {
+        EntityProfile saved = saveRandomEntityProfile(randomAlphanumeric(10), EntityType.TENANT);
+        PageLink pageLink = new PageLink(1);
+        List<EntityProfile> data = profileService.findTenantProfiles(pageLink).getData();
+        assertEquals(pageLink.getPageSize(), data.size());
+        assertEquals(saved, data.get(0));
+    }
+
+    @Test
+    public void testFindById() {
+        EntityProfile saved = saveRandomEntityProfile();
+        assertEquals(saved, profileService.findById(TenantId.SYS_TENANT_ID, saved.getId()));
+    }
+
+    @Test(expected = DataIntegrityViolationException.class)
+    public void testSaveWithSameNameTenantAndType() {
+        String sameName = randomAlphanumeric(10);
+        EntityType sameEntityType = EntityType.DEVICE;
+        saveRandomEntityProfile(sameName, sameEntityType);
```

Comment:
Let's also check that sameName but different entityTypes works fine

Reply:
added